Bottom: 279fbcbda26f35a3eadf99f2b2339cdcc96a1180
Top:    c429faa1d9cb8633a0d821bc7b104a661e3d1063
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 17:23:47 +0100

Add missing HID structure zero initialisation for gcode and png HIDs.

Missed from commit commit d67f4335b5d4f97330448b833ea1bf04377c3a57

  "Initialise the HID vtable by function, rather than as a struct initialiser."

I missed the memset from these two HIDs.


---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 9360cb7..3e49ed3 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1035,6 +1035,8 @@ HID gcode_hid;
 void
 hid_gcode_init ()
 {
+  memset (&gcode_hid, 0, sizeof (HID));
+
   gcode_hid.struct_size         = sizeof (HID);
   gcode_hid.name                = "gcode";
   gcode_hid.description         = "G-CODE export.";
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index ca87a84..7c2f461 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1519,6 +1519,8 @@ png_set_crosshair (int x, int y, int a)
 void
 hid_png_init ()
 {
+  memset (&png_hid, 0, sizeof (HID));
+
   png_hid.struct_size = sizeof (HID);
   png_hid.name        = "png";
   png_hid.description = "GIF/JPEG/PNG export.";
